src/Mvc/Mvc.Core/src/Formatters/FormatFilter.cs

```diff
@@ -155,14 +154,14 @@ public void OnResultExecuting(ResultExecutingContext context)
                 return;
             }
 
-            if (!(context.Result is ObjectResult objectResult))
+            if (context.Result is not ObjectResult objectResult)
```

Comment:
What type is objectResult here? "not ObjectResult" seems like multiple types

Reply:
Yeah, it's supposed to be any other IActionResult type. For instance, if your action does:

```
if (id == 0)
{
   return NotFound();
}

return Ok(someObject);
```

We want the format to only apply to the latter. It's very specific.